src/OrchardCore/OrchardCore.Setup.Core/SetupService.cs

```diff
@@ -170,33 +184,31 @@ await scope
                 }
 
                 executionId = Guid.NewGuid().ToString("n");
+            }
 
-                // Create a new scope for the recipe thread to prevent race issues with other scoped
-                // services from the request.
-                using (var scope = shellContext.CreateScope())
-                {
-                    var recipeExecutor = scope.ServiceProvider.GetService<IRecipeExecutor>();
+            var recipeExecutor = new RecipeExecutor(
```

Comment:
why are we not relying on the DI like we used to? Seems weird to create a new one manually.

Reply:
Using DI here means that we need to keep the previous container alive, but the `RecipeExecutor` will have to reload the tenant it's setting up, creating a new container. And it can do it multiple times. Because of this we arrive we two containers for the same tenant running at the same time, and the components that "require" DI will break when they don't expect that. In this case it's the data access layer using SQlite, as it goes into concurrency issues, two `Store` are created, and try to maintain incompatible states in the same file. And if this happens for this component, this will happen for others.

The solution is just to close the previous container and not have `RecipeExecutor` be created by DI. It works because we define its scope ourselves and we give it the required services from the application scope.

I did it while pair programming with @jtkech so it's not just me who came to this conclusion.